ext/reflection/php_reflection.c

```diff
@@ -1372,33 +1330,81 @@ static parameter_reference *_reflection_param_get_default_param(INTERNAL_FUNCTIO
 	}
 
 	param = intern->ptr;
-	if (param->fptr->type != ZEND_USER_FUNCTION) {
-		zend_throw_exception_ex(reflection_exception_ptr, 0, "Cannot determine default value for internal functions");
-		return NULL;
-	}
 
 	return param;
 }
 /* }}} */
 
-/* {{{ _reflection_param_get_default_precv */
-static zend_op *_reflection_param_get_default_precv(INTERNAL_FUNCTION_PARAMETERS, parameter_reference *param)
+/* {{{ _reflection_param_get_default_arg_info */
+static int _reflection_param_get_default_arg_info(zend_arg_info *arg_info, zval *default_value_zval)
```

Comment:
For userland functions, we should still try to fetch the already evaluated value from the RECV_INIT opcode here, to avoid a much more expensive runtime evaluation for that case.

Reply:
I reimplemented fetching from `RECV_INIT` for the `getDefaultValue()` method.